In [58]:
import tensorflow as tf
import json
from tensorflow.keras.preprocessing.text import Tokenizer
from tensorflow.keras.preprocessing.sequence import pad_sequences

In [72]:
dataset = []
sentences = []
labels = []
urls = []
for line in open('data/sarcasm.json', 'r'):
    dataset.append(json.loads(line))
    
for item in dataset:
    sentences.append(item['headline'])
    labels.append(item['is_sarcastic'])
    urls.append(item['article_link'])

In [73]:
tokenizer = Tokenizer(oov_token="<OOV>")
tokenizer.fit_on_texts(sentences)
word_index = tokenizer.word_index

seqences = tokenizer.texts_to_sequences(sentences)
padded = pad_sequences(seqences, padding='post')
print(padded[0])
print(padded.shape)

[  308 15115   679  3337  2298    48   382  2576 15116     6  2577  8434
     0     0     0     0     0     0     0     0     0     0     0     0
     0     0     0     0     0     0     0     0     0     0     0     0
     0     0     0     0]
(26709, 40)


#### We've got 26709 sentences each with 40 tokens
#### Let's split our data into training and testing sets

In [76]:
train_sentences = sentences[0:10000]
test_sentences = sentences[10000:]
train_labels = labels[0:10000]
test_labels = labels[10000:]